File: app/src/main/java/com/organizer/MainActivity.java
package com.organizer;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import java.util.Objects;
import com.organizer.layouts.MainLayout;

public class MainActivity extends AppCompatActivity
{
    private static MainActivity activity;
    private static DisplayMetricsController displayMetricsController;
    private static DateController dateController;
    private MainLayout mainLayout;
    
    public MainActivity()
    {
        activity = this;
    }
    
    public static MainActivity getInstance()
    {
        return activity;
    }
    
    public static DisplayMetricsController getDisplayMetricsController()
    {
        return displayMetricsController;
    }
    
    public static DateController getDateController()
    {
        return dateController;
    }
    
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        Objects.requireNonNull(this.getSupportActionBar()).hide();
        
        displayMetricsController = new DisplayMetricsController(getWindowManager(), getResources().getDisplayMetrics().density);
        dateController = new DateController();
        
        mainLayout = new MainLayout(this);
        setContentView(mainLayout);
        
        new Task(this, dateController.buildId(dateController.todayD, dateController.todayM, dateController.todayY), "Work", 8, 12);
        new Task(this, dateController.buildId(dateController.todayD + 1, dateController.todayM, dateController.todayY), "Programming", 6, 8);
        new Task(this, dateController.buildId(dateController.todayD + 1, dateController.todayM, dateController.todayY), "Dancing", 9, 15);
        new Task(this, dateController.buildId(dateController.todayD + 2, dateController.todayM, dateController.todayY), "Work", 6, 17);
        new Task(this, dateController.buildId(dateController.todayD + 2, dateController.todayM, dateController.todayY), "Family Dinner", 17, 18);
        new Task(this, dateController.buildId(dateController.todayD + 3, dateController.todayM, dateController.todayY), "Dancing", 6, 24);
    }
    
    @Override
    protected void onResume()
    {
        super.onResume();
        //mainLayout.getToDoLayout().init();
    }
    
    public MainLayout getLayout()
    {
        return mainLayout;
    }
}

File: app/src/main/kotlin/com/organizer/MainActivity.kt
package com.organizer

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.organizer.layouts.MainLayout
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin
import org.koin.dsl.module
import org.koin.java.KoinJavaComponent

class MainActivity : AppCompatActivity() {

    companion object {
        inline fun <reified T> inject(): Lazy<T> = KoinJavaComponent.inject(T::class.java)

        inline fun <reified T> injectNow(): T = KoinJavaComponent.inject<T>(T::class.java).value
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.hide()

        startKoin {
            androidLogger()
            androidContext(this@MainActivity)
            modules(module {
                single { this@MainActivity }
                single { DisplayMetricsController(windowManager, resources.displayMetrics.density) }
                @Suppress("USELESS_CAST")
                single { PreferenceServiceImpl(getPreferences(MODE_PRIVATE)) as PreferenceService }
                single { DateController() }
                single { MainLayout() }
            })
        }
        setContentView(injectNow<MainLayout>())
        val dateController = injectNow<DateController>()

        CalendarEvent(this, dateController.buildId(dateController.todayD, dateController.todayM, dateController.todayY), "Work", 8.0, 12.0)
        CalendarEvent(this, dateController.buildId(dateController.todayD + 1, dateController.todayM, dateController.todayY), "Programming", 6.0, 8.0)
        CalendarEvent(this, dateController.buildId(dateController.todayD + 1, dateController.todayM, dateController.todayY), "Dancing", 9.0, 15.0)
        CalendarEvent(this, dateController.buildId(dateController.todayD + 2, dateController.todayM, dateController.todayY), "Work", 6.0, 17.0)
        CalendarEvent(this, dateController.buildId(dateController.todayD + 2, dateController.todayM, dateController.todayY), "Family Dinner", 17.0, 18.0)
        CalendarEvent(this, dateController.buildId(dateController.todayD + 3, dateController.todayM, dateController.todayY), "Dancing", 6.0, 24.0)
    }

//    override fun onResume() {
//        super.onResume()
//        mainLayout.getToDoLayout().init();
//    }
}
